app/main.py

```
@@ -1,34 +1,89 @@
+        elif start[0] == end[0]:
+            for column_number in range(start[1], end[1] + 1):
+                self.decks.append(Deck(start[0], column_number))
+        elif start[1] == end[1]:
+            for row_number in range(start[0], end[0] + 1):
+                self.decks.append(Deck(row_number, start[1]))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for desk in self.decks:
+            if desk.row == row and desk.column == column:
+                return desk
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> str:
+        fire_desk = self.get_deck(row, column)
+        fire_desk.is_alive = False
+        for desk in self.decks:
+            if desk.is_alive:
+                return "Hit!"
+        self.is_drowned = True
+        return "Sunk!"
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def check_neighbor_cells(self, ship: Ship, ship_desk: tuple) -> None:
+        for row in range(ship_desk[0] - 1, ship_desk[0] + 2, 2):
+            for column in range(ship_desk[1] - 1, ship_desk[1] + 2, 2):
+                if ((row, column) in self.field) \
+                        and (self.field[(row, column)] != ship):
+                    print("ships shouldn't be located "
```

use raise exception instead of print